cli/cfncluster/cfncluster.py

```diff
@@ -403,6 +413,54 @@ def instances(args):
     for instance in instances:
         print('%s         %s' % (instance[0],instance[1]))
 
+def get_head_ip(outputs):
+    for output in outputs:
+        if output.key == "MasterPublicIP":
+            return output.value
+
+def get_head_user(parameters):
+    base_os = [i.value for i in parameters if i.key == "BaseOS"][0]
+    return username_map[base_os]
+
+def login(args):
+    stack = ('cfncluster-' + args.cluster_name)
+    config = cfnconfig.CfnClusterConfig(args)
+    cfnconn = boto.cloudformation.connect_to_region(config.region,aws_access_key_id=config.aws_access_key_id,
+                                                    aws_secret_access_key=config.aws_secret_access_key)
+    try:
+        status = cfnconn.describe_stacks(stack)[0].stack_status
+        if not ((status == 'CREATE_COMPLETE') or (status == 'UPDATE_COMPLETE')):
+            logger.info("Status: %s. Needs to be in CREATE_COMPLETE or UPDATE_COMPLETE." % (status))
+            sys.exit(1)
+        state = poll_master_server_state(stack, config, args.nowait)
+        if state != 'running':
+            logger.info("MasterServer: %s. Needs to be running to login." % (state.upper))
+            sys.exit(1)
+        key_path = ""
+        if config.key_path is not None:
+            key_path = "-i %s" % config.key_path
+        if args.key_path is not None:
+            key_path = "-i %s" % args.key_path
+        stack_result = cfnconn.describe_stacks(stack)[0]
```

Comment:
In case key_path is empty then throw error message

Reply:
The idea is to make it optional for customers who have their keys in the default `~/.ssh` folder